front/app/services/projectFolders.ts

```diff
@@ -1,11 +1,30 @@
 import { Multiloc, ImageSizes, IRelationship } from 'typings';
-
 import streams, { IStreamParams } from 'utils/streams';
 import { isNilOrError } from 'utils/helperUtils';
-
 import { API_PATH } from 'containers/App/constants';
-import { PublicationStatus } from 'services/projects';
+import {
+  PublicationStatus,
+  CARD_IMAGE_ASPECT_RATIO as PROJECT_CARD_IMAGE_ASPECT_RATIO,
+} from 'services/projects';
+import { TDevice } from 'components/admin/SelectPreviewDevice';
+import { TProjectFolderCardSize } from 'components/ProjectAndFolderCards/components/ProjectFolderCard';
+
 const apiEndpoint = `${API_PATH}/project_folders`;
+export const CARD_IMAGE_ASPECT_RATIO = PROJECT_CARD_IMAGE_ASPECT_RATIO;
+export const getCardImageUrl = (
+  imageVersions: ImageSizes,
+  device: TDevice,
+  size?: TProjectFolderCardSize
+) => {
+  if (device === 'phone') {
```

Comment:
It would be very convenient to detect the device type right in this function.
1. We won't need to call `const device = useDevice();` in the component
2. We won't need to pass it in this function.

Currently, it's not possible because `getCardImageUrl` is not a hook (and it seems transforming it into a hook would make it inconvenient to use). But we can create a function `getBreakpoint('phone')` that returns true/false and can be used here.

3. We won't need `useDevice` anymore.

But it's quite some refactoring (releasing `cl-components-library` too), so I cannot insist.

Reply:
`useBreakpoint` is a hook for a good reason - it requires an event listener that needs to be cleaned up when a component unmounts. So it's tied to the lifecycle of the component that it's being used in. However, `useDevice` does not really need to be a React hook - it can just be a helper function where you pass the data from `useBreakpoint` and it returns the device. 

Something like:
```
const getDevice = ({ isPhone, isTablet }) => {
    if (isPhone) return 'phone';
    if (isTablet) return 'tablet';
    return 'desktop';
}
```

Then you can use this in your component, instead of the `useDevice` hook. However, this would mean you need to use `useBreakpoint` twice in your component every time you want to use this method, so perhaps it's still preferable to abstract this logic in a `useDevice` hook. 

Not that I think about it, the name is a bit confusing though, as you are not really detecting the device you are on, just the width of the screen. Someting like `useBreakpointName` might be more accurate (or if you can come up with a better name ever better) but I'm not insisting on it.

Reply:
Also, may I suggest we use a native way to handle this, instead of managing it manually? Our component library `Image` component supports `sizes` prop which will do this for you out of the box without the need of any hooks / helpers. More info [here](https://www.dofactory.com/html/img/sizes).

Reply:
@IvaKop 

I've tried the version with getBreakpoint. It uses the code from useBreakpoint and seems the simplest option: https://github.com/CitizenLabDotCo/citizenlab/pull/3828. I would still move this to the component library.

There's no need to call useBreakpoint when we use useDevice. It seems to work without the event listener as well (getBreakpoint gets called again on resizing the screen). Wdyt?

Reply:
I'm not sure I understand what you mean here. `useDevice` calls `useBreakpoint` internally. For `getBreakpoint` to be called again, it means something is causing the component to re-render of screen size change. Let's have a chat about it to make sure I understand the issue properly.